Add Dart support

techee opened the pull request:
This PR adds support for the Dart language:
1. It uses the Dart lexer which has been recently merged to lexilla
2. The keywords are based on Notepad4 keywords
3. Build menu settings is based on https://github.com/geany/geany/pull/3372

b4n commented on scintilla/scintilla_changes.patch:
```diff
@@ -84,6 +84,7 @@ index cd4b23617..af4a73db4 100644
 +		&lmCPP,
 +		&lmCss,
 +		&lmD,
++		&lmDart,
```
As this isn't updating the chunk header, it's gonna corrupt the patch, isn't it?

b4n replied:
…well OK, it's already broken since at least a9363bc15d2c66ea91eeec923961f93e41bff216, but is that an excuse? :)

techee replied:
Yeah, sorry, I've been blindly following the instructions in HACKING without thinking much for quite some time ;-).

I've just created https://github.com/geany/geany/pull/3991 to better describe how the patch file should be modified when adding a lexer.

b4n reviewed (commented):
LGTM from little review and testing apart from:
* the comment about the patch file
* the absence of tags parsing.  the custom filetype version uses Java, which seems to do a decent job (beware that not knowing the language, I just copied a couple of samples and lanced at the results), should this one do the same?

techee commented:
> the absence of tags parsing. the custom filetype version uses Java, which seems to do a decent job (beware that not knowing the language, I just copied a couple of samples and lanced at the results), should this one do the same?

This is what I wanted to test but then forgot about it. I'll add it.

techee commented:
@b4n I've rebased this PR on top of master, resolved conflicts and updated build commands based on https://github.com/geany/geany/issues/371#issuecomment-2408607518 (I didn't receive any feedback there but I think it can be easily modified if users report problems with them).

b4n reviewed (approved):
LGBI and compared to previous version should also work fine 👍
